easy-ui-react/vite.config.ts

```diff
@@ -0,0 +1,58 @@
+/// <reference types="vitest" />
+
+import react from "@vitejs/plugin-react";
+import glob from "glob";
+import { defineConfig } from "vite";
+import { viteStaticCopy } from "vite-plugin-static-copy";
+
+export default defineConfig({
+  plugins: [
+    react({ jsxRuntime: "classic" }),
+    viteStaticCopy({
+      targets: [
+        { src: "package.json", dest: ".", transform: transformPkgJson },
+      ],
+    }),
+  ],
+  css: {
+    modules: {
+      globalModulePaths: [/global\.s?css$/],
+    },
+  },
+  build: {
+    emptyOutDir: false,
+    lib: {
+      entry: buildEntryObject([
+        ...glob.sync("src/**/index.ts"),
+        ...glob.sync("src/utilities/*.ts"),
+      ]),
+    },
+    rollupOptions: {
+      external: ["react", "react-dom"],
+    },
+  },
+  test: {
+    globals: true,
+    environment: "jsdom",
+    css: true,
+    setupFiles: "./vitest.setup.ts",
+  },
+});
+
+// Maps entry filenames to entries with proper aliases:
+//
+// src/Placeholder/index.ts -> Placeholder/index
+// src/utilities/css.ts -> utilities/css
+function buildEntryObject(entries: string[]) {
+  return entries.reduce((o, entry) => {
+    const alias = entry.replace(/^src\//, "").replace(/.ts$/, "");
+    return { ...o, [alias]: entry };
+  }, {});
+}
+
+// Cleans up package.json file for including in distribution
+function transformPkgJson(code) {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { scripts, devDependencies, exports, ...restPkg } = JSON.parse(code);
+  return JSON.stringify(restPkg, null, 2);
+}
```

Comment:
out of curiosity, why was the config file moved into easy-ui-react?

Reply:
i originally created it in the root because i thought Storybook used it, but it actually doesn't lol. it uses our Vite instance but it has its own config (by default). so then i realized only our easy-ui-react project is using a Vite config so i just moved it in there for now. it's possible we'll have other projects in the future that we'll warrant us sharing config across projects in which case it'd probably make sense to have a root one at that point